executor/builder.go

```diff
@@ -1800,7 +1800,10 @@ func (b *executorBuilder) buildIndexLookUpJoin(v *plannercore.PhysicalIndexJoin)
 	e.innerCtx.keyCols = innerKeyCols
 	e.joinResult = newFirstChunk(e)
 	executorCounterIndexLookUpJoin.Inc()
-	return e
+	if v.KeepOuterOrder {
```

Comment:
Do we need to modify the cost model of index join when `KeepOuterOrder` is false?

Reply:
Yes, we need~
It's a separated PR, the review can be focused on the execution phase now.